<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>cloud_function</class>
 <widget class="QMainWindow" name="cloud_function">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>500</height>
   </rect>
  </property>
  <property name="font">
   <font>
    <family>Microsoft YaHei UI</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>云函数部署程序打包工具</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_12">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3" stretch="0,1">
      <item>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="title">
         <string>打包环境与项目路径</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="leftMargin">
          <number>4</number>
         </property>
         <property name="topMargin">
          <number>4</number>
         </property>
         <property name="rightMargin">
          <number>4</number>
         </property>
         <property name="bottomMargin">
          <number>4</number>
         </property>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <property name="spacing">
            <number>2</number>
           </property>
           <item>
            <widget class="QLabel" name="label">
             <property name="text">
              <string>选择打包环境(调用 pip 模块)：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="uiComboBox_python_envs"/>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="spacing">
            <number>2</number>
           </property>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>云函数项目目录路径：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="uiComboBox_preject_path">
             <property name="editable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QCheckBox" name="uiCheckBox_confirm_project_path">
               <property name="text">
                <string>确认项目路径</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="uiPushButton_clear_path">
               <property name="text">
                <string>清除</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="uiPushButton_select_projectdir">
               <property name="text">
                <string>选择</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>云函数项目的依赖清单（requirements.txt）</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="leftMargin">
          <number>4</number>
         </property>
         <property name="topMargin">
          <number>4</number>
         </property>
         <property name="rightMargin">
          <number>4</number>
         </property>
         <property name="bottomMargin">
          <number>4</number>
         </property>
         <item>
          <widget class="QCheckBox" name="uiCheckBox_upgrade_requires">
           <property name="text">
            <string>每次打包都选择重装和升级现有依赖包</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QCheckBox" name="uiCheckBox_overwrite_requirement">
           <property name="text">
            <string>打包时以表格依赖项覆盖依赖清单文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QTableWidget" name="uiTableWidget_reqirement_lines">
           <property name="selectionMode">
            <enum>QAbstractItemView::SingleSelection</enum>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <property name="cornerButtonEnabled">
            <bool>false</bool>
           </property>
           <attribute name="horizontalHeaderDefaultSectionSize">
            <number>25</number>
           </attribute>
           <attribute name="horizontalHeaderHighlightSections">
            <bool>false</bool>
           </attribute>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>项目依赖</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>安装结果</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string/>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <spacer name="horizontalSpacer_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QPushButton" name="uiPushButton_delete_line">
             <property name="palette">
              <palette>
               <active>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>0</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
               </active>
               <inactive>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>255</red>
                   <green>0</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
               </inactive>
               <disabled>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>120</red>
                   <green>120</green>
                   <blue>120</blue>
                  </color>
                 </brush>
                </colorrole>
               </disabled>
              </palette>
             </property>
             <property name="text">
              <string>删除行</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="uiPushButton_add_line">
             <property name="text">
              <string>添加行</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_14" stretch="0,0,0">
      <item>
       <widget class="QGroupBox" name="groupBox_5">
        <property name="title">
         <string>打包配置储存与切换</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_10">
         <property name="spacing">
          <number>4</number>
         </property>
         <property name="leftMargin">
          <number>4</number>
         </property>
         <property name="topMargin">
          <number>4</number>
         </property>
         <property name="rightMargin">
          <number>4</number>
         </property>
         <property name="bottomMargin">
          <number>4</number>
         </property>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_9">
           <property name="spacing">
            <number>2</number>
           </property>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>保存当前打包配置(点击下方列表快速填充)</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="1,0">
             <item>
              <widget class="QLineEdit" name="uiLineEdit_config_name">
               <property name="placeholderText">
                <string>新的打包配置</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="uiPushButton_save_config">
               <property name="text">
                <string>保存</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_8" stretch="0,1">
           <property name="spacing">
            <number>2</number>
           </property>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>已保存的打包配置</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,0">
             <item>
              <widget class="QListWidget" name="uiListWidget_config_list"/>
             </item>
             <item>
              <layout class="QVBoxLayout" name="verticalLayout_7">
               <item>
                <widget class="QPushButton" name="uiPushButton_switch_config">
                 <property name="text">
                  <string>切换</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="uiPushButton_del_scfconfig">
                 <property name="palette">
                  <palette>
                   <active>
                    <colorrole role="ButtonText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>255</red>
                       <green>0</green>
                       <blue>0</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </active>
                   <inactive>
                    <colorrole role="ButtonText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>255</red>
                       <green>0</green>
                       <blue>0</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </inactive>
                   <disabled>
                    <colorrole role="ButtonText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>120</red>
                       <green>120</green>
                       <blue>120</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </disabled>
                  </palette>
                 </property>
                 <property name="text">
                  <string>删除</string>
                 </property>
                </widget>
               </item>
               <item>
                <spacer name="verticalSpacer_2">
                 <property name="orientation">
                  <enum>Qt::Vertical</enum>
                 </property>
                 <property name="sizeHint" stdset="0">
                  <size>
                   <width>20</width>
                   <height>40</height>
                  </size>
                 </property>
                </spacer>
               </item>
              </layout>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_3">
        <property name="title">
         <string>依赖包安装位置（先安装后打包）</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_6">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="leftMargin">
          <number>4</number>
         </property>
         <property name="topMargin">
          <number>4</number>
         </property>
         <property name="rightMargin">
          <number>4</number>
         </property>
         <property name="bottomMargin">
          <number>4</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout" stretch="0,0,1">
           <property name="spacing">
            <number>4</number>
           </property>
           <item>
            <widget class="QRadioButton" name="uiRadioButton_using_autotempdir">
             <property name="text">
              <string>临时文件夹</string>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QRadioButton" name="uiRadioButton_using_projectdir">
             <property name="text">
              <string>项目文件夹</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>10</width>
               <height>10</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="0,1,0">
           <property name="spacing">
            <number>4</number>
           </property>
           <item>
            <widget class="QRadioButton" name="uiRadioButton_using_customtemp">
             <property name="text">
              <string>自定义</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="uiLineEdit_customdir_path"/>
           </item>
           <item>
            <widget class="QPushButton" name="uiPushButton_select_customdir">
             <property name="text">
              <string>选择</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="groupBox_4">
        <property name="title">
         <string>保存位置与开始打包</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_11">
         <property name="spacing">
          <number>2</number>
         </property>
         <property name="leftMargin">
          <number>4</number>
         </property>
         <property name="topMargin">
          <number>4</number>
         </property>
         <property name="rightMargin">
          <number>4</number>
         </property>
         <property name="bottomMargin">
          <number>4</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_8">
           <property name="spacing">
            <number>4</number>
           </property>
           <item>
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>生成的文件名</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="uiLineEdit_generatedname"/>
           </item>
           <item>
            <widget class="QCheckBox" name="uiCheckBox_overwrite_samefile">
             <property name="text">
              <string>覆盖同名文件</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_9">
           <property name="spacing">
            <number>4</number>
           </property>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>文件保存位置</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="uiLineEdit_generateddir"/>
           </item>
           <item>
            <widget class="QPushButton" name="uiPushButton_select_generateddir">
             <property name="text">
              <string>选择</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_11">
           <property name="spacing">
            <number>4</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_12">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="0,1">
               <property name="spacing">
                <number>4</number>
               </property>
               <item>
                <widget class="QLabel" name="uiLabel_working_movie">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="uiLabel_working_tips">
                 <property name="text">
                  <string/>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="0,1">
               <property name="spacing">
                <number>8</number>
               </property>
               <item>
                <widget class="QLabel" name="uiLabel_whatis_cloudfunction">
                 <property name="palette">
                  <palette>
                   <active>
                    <colorrole role="WindowText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>0</red>
                       <green>0</green>
                       <blue>255</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </active>
                   <inactive>
                    <colorrole role="WindowText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>0</red>
                       <green>0</green>
                       <blue>255</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </inactive>
                   <disabled>
                    <colorrole role="WindowText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>120</red>
                       <green>120</green>
                       <blue>120</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </disabled>
                  </palette>
                 </property>
                 <property name="font">
                  <font>
                   <family>Microsoft YaHei UI</family>
                   <pointsize>8</pointsize>
                  </font>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string>什么是云函数？</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="uiLabel_set_excludes">
                 <property name="palette">
                  <palette>
                   <active>
                    <colorrole role="WindowText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>0</red>
                       <green>0</green>
                       <blue>255</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </active>
                   <inactive>
                    <colorrole role="WindowText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>0</red>
                       <green>0</green>
                       <blue>255</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </inactive>
                   <disabled>
                    <colorrole role="WindowText">
                     <brush brushstyle="SolidPattern">
                      <color alpha="255">
                       <red>120</red>
                       <green>120</green>
                       <blue>120</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </disabled>
                  </palette>
                 </property>
                 <property name="font">
                  <font>
                   <family>Microsoft YaHei UI</family>
                   <pointsize>8</pointsize>
                  </font>
                 </property>
                 <property name="cursor">
                  <cursorShape>PointingHandCursor</cursorShape>
                 </property>
                 <property name="text">
                  <string>排除目录和文件</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QPushButton" name="uiPushButton_start_scfpacking">
             <property name="minimumSize">
              <size>
               <width>100</width>
               <height>36</height>
              </size>
             </property>
             <property name="palette">
              <palette>
               <active>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>0</red>
                   <green>170</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
               </active>
               <inactive>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>0</red>
                   <green>170</green>
                   <blue>0</blue>
                  </color>
                 </brush>
                </colorrole>
               </inactive>
               <disabled>
                <colorrole role="ButtonText">
                 <brush brushstyle="SolidPattern">
                  <color alpha="255">
                   <red>120</red>
                   <green>120</green>
                   <blue>120</blue>
                  </color>
                 </brush>
                </colorrole>
               </disabled>
              </palette>
             </property>
             <property name="font">
              <font>
               <family>Microsoft YaHei UI</family>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>开始打包</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
